test/watcher_test.js

```diff
@@ -3,8 +3,35 @@ import chai from 'chai';
 import sinonChai from 'sinon-chai';
 import Sinon from 'sinon';
 import Watcher from '../lib/watcher';
+import broccoli from '..';
+const Builder = broccoli.Builder;
+const multidepRequire = require('multidep')('test/multidep.json');
 import SourceNodeWrapper from '../lib/wrappers/source-node';
 
+const Plugin = multidepRequire('broccoli-plugin', '1.3.0');
+
+function defer() {
+  let deferred = {};
+  let promise = new Promise(function(resolve, reject) {
+    deferred.resolve = resolve;
+    deferred.reject = reject;
+  });
+  deferred.promise = promise;
+  return deferred;
+}
+
+function waitForWatcherToBeReady(watcher) {
+  return new Promise(resolve => {
+    if (watcher._ready === true) {
```

Comment:
Using private API here isn't ideal. Maybe we should simply add this method to the watcher itself?

Reply:
Added it to the watcher itself.